src/vehicle.cpp

```diff
@@ -1529,6 +1529,7 @@ void VehicleEnterDepot(Vehicle *v)
 	}
 	SetWindowDirty(WC_VEHICLE_VIEW, v->index);
 
+	DepotID depot_id = GetDepotIndex(v->tile);
```

Comment:
Why determine it here, when it's only used in one place later on within an if-statement and all kinds of other checks? Seems to me it's only needed when coming to that particular comparison, or does this have important side effects?

Reply:
On subsequent commits of #9577, depot_id is to be used more times. In fact, when the depot window numbers are changed, depot_id is used in line 1536 and probably more times through these lines.
As I have been rebasing this changes for some time, it was easier to store the depot_id here to resolve conflicts.
I can do as you suggest, but it is just adding more work.

Just let me know whether I should do the change you proposed.